packages/website/src/components/options.ts

```diff
@@ -44,4 +44,6 @@ export const defaultConfig: ConfigModel = {
   }),
   scroll: true,
   showTokens: false,
+  esQueryFilter: '',
+  esQuerySelector: undefined,
```

Comment:
I don't really like the fact that I have 2 variables that basically hold the same state (string vs selector object) instead of having one of them, and the other will be a derived state.
I wasn't able to find an ESQuery method that converts a selector to a string, seems to be a one-way ticket. It means that I can save only the string and then parse it to a selector. I tried 2 approaches:

1. Parse when reading the initial state from the URL - Didn't work because `window.esquery` isn't available yet
2. Passing the string as a default value to the `ESQueryFilter` input and parse when on mount (using the same logic of the `onChange` handler) - Looked ugly and felt wrong (`useEffect` for such cases always feels to me like a code smell)

Any idea?

Reply:
Yeah +1 to not having two variables. Also +1 that `useEffect` feels like a code smell for this case. Tricky... 🤔 

The first thought that comes to my head is changing the state to be `string | undefined | ESQueryFilter`. It'd be `string | undefined` when first initialized, then turn into `ESQueryFilter` later... I don't love that. Seems kind of convoluted.

Second thought: how about ignoring the return of `window.esquery.parse(value)` inside `ESQueryFilter` and using a second `window.esquery.parse(filter)` inside `ASTViewer`'s `tryToApplyFilter`? It's a duplicate parse but removes a state...

Third thought: this also feels like a reasonable feature request on ESQuery. Up for filing an issue there? 😄 

Fourth thought: maybe instead of a duplicate parse, the state should be an object like `{ filter: string; selector?: ESQuery.Selector } | undefined`? In my mind that's a better representation of what the state truly is.

Reply:
> Do you mean that instead of passing a selector we'll now pass a filter and the AST viewer will handle the parsing?

Exactly, yeah! Either this or option 4 make me less unhappy.